edgecontext.thrift

```diff
@@ -126,7 +131,11 @@ struct Locale {
     /** IETF language code representing the client locale preferences.
     Can be either {lang} or {lang}_{region} format. e.g. en, en_US
     */
-    1: LocaleCode locale_code
+    1: LocaleCode locale_code;
+    /** Locale code forced to the BCP-47 format (hyphen separated e.g de-DE, pt-BR, etc.).
+    This field is introduced to be used for localization instead of locale_code.
+    */
+    2: UnifiedLocaleCode unified_locale_code;
```

Comment:
`locale_code` can be either `en` or `en_US`, and the regexp of this one also supports both `en` or `en-US`, so it looks to me that this is just a different format of `locale_code`, but not provide any additional information? why do we need both?

Reply:
Yeah, this is just a different format of `locale_code`. We are going to use it across all services for localization. 
This is suitable for several reasons:

- Golang supported this format out of the box ([link](https://pkg.go.dev/golang.org/x/text/language)). And we've already declared supported locales in this format for golang services (e.g. [survey](https://github.snooguts.net/reddit/reddit-service-survey/blob/main/service/survey.go#L23))

- Both our localization libraries (for [python](https://github.snooguts.net/reddit/i18n-tools-py) and [golang](https://github.snooguts.net/reddit-go/i18n)) can work with this format. So, we can update services to use this format (e.g. [1](https://github.snooguts.net/reddit/reddit-service-graphql/pull/5308/files), [2](https://github.snooguts.net/reddit/reddit-service-mailroom/pull/1930/files), [3](https://github.snooguts.net/reddit/reddit-service-accountmanager/pull/1074))

The `locale_code` is saved for backward compatibility until we update all services to use the new field. Then will discuss and probably remove the old one.

Reply:
@ZackMcBride pavel's explanation here is that essentially `locale_code` is deprecated in favor of the new `unified_locale_code`.

if that's not the case, then we go back to my original question here: `unified_locale_code` is essentially a superset of `locale_code` (just use slightly different format, `-` instead of `_`), why do we keep both?

Reply:
Sorry missed this part of the conversation. My understanding is that `unified_locale_code` is not just a reformatted version of `locale_code`, but that we're also going to use this field to store the negotiated best match locale when compared to some list of supported locales.

For example, if `fr-CA` is supplied by a client, but is "unsupported", then we would negotiate that locale to a best match, e.g. `fr-FR`. In that case `locale_code` would contain the original one, and `unified_locale_code` would contain the correctly formatted, "supported", one. 

I believe we want to keep track of the original locale for accurate user preferences. I'd like for us to follow up on this though to figure out if it's necessary.

Reply:
in that case the name of the field should reflect that (something like `negotiated_locale_code`)